include/deal.II/lac/solver.h

```diff
@@ -120,6 +120,24 @@ template <typename number> class Vector;
  * <tt>swap(VECTOR &a, VECTOR &b)</tt> that exchanges the values of the two
  * vectors.
  *
+ * Finally, the solvers also expect an instantiation of
+ * GrowingVectorMemory<VECTOR>. These instantiations are provided by the
+ * deal.II library for the built-in vector types, but must be explicitly added
+ * for user-provided vector classes. Otherwise, the linker will complain that
+ * it cannout find the constructors and destructors of GrowingVectorMemory
+ * that happen in the @p Solver class below.
+ *
+ * @code
+ * #include <deal.II/lac/vector_memory.templates.h>
+ *
+ * // Definition and implementation of vector class
+ * class UserVector { ... };
+ *
+ * // Create explicit instantiation for the vector class
+ * template class VectorMemory<UserVector >;
+ * template class GrowingVectorMemory<UserVector >;
```

Comment:
This will lead to multiple definition errors if it happens in a `.h` file (where one would typically have the class declaration of `UserVectro`) that is included from multiple locations.  Maybe a note to put these two lines into one of the `.cc` files?